cli/mfg.c

```diff
@@ -22,6 +22,8 @@
  *
  */
 
+#ifdef __linux__
+
```

Comment:
Why?  Is this really necessary? I don't see any good reason to disable this except "recommendations".

At worst, I'd suggest printing a warning in Windows and say it's not supported. Not building it entirely is just a pain given that there's no good reason why it wouldn't work.

If you really must do this,put a gigantic comment explaining why and preferably a much more detailed commit message why you're doing this.

Reply:
'mfg' commands are used for configuring security settings and recovering from boot failures, and are most often used with an I2C device. While Linux provides a unified I2C abstraction layer, making it easy to support different I2C adapters, there is no equivalent I2C abstraction layer on Windows, so it becomes extremely difficult to support different I2C drivers. Due to this reason, the 'mfg' commands are removed from non-Linux build.

Reply:
Creating a hard rule based on a "most often" statement seems silly to me.

If the commands can be done over PCI then allow them to be done over PCI. 

If they can only be done over I2C then encode that rule which has nothing to do with Linux/Windows. 

But don't create arbitrary rules based on weird expectations.

Reply:
I agree with your points. From a technical perspective, we should fix the problem at the root, instead of creating a build rule that masks it.

However, the decision to remove Windows build is more business-related. The Windows customer base and the current development resource we have do not warrant supporting Windows build, plus we are short in testing/verification resources as well. As the 'mfg' commands have potential to brick a device if used incorrectly, it is not prudent to release the Windows tool without sufficient testing. It is based on these factors that the Windows build was dropped from release roadmap.

Having said that, I would be happy to find a middle ground where the technical and business considerations can meet. Feel free to propose other solutions if any.

Reply:
Well, like I said, add a *warning* saying that these commands are not supported by Microsemi in Windows. Because that's what is actually happening. You could make it prompt with a Y/N continue or something like that.

Reply:
I have updated the code to check for OS in `mfg_handler()`, print an error then exit. 

I think it is better to exit the program than continuing, since the command have not been tested on Windows.

Reply:
@lsgunth please take a look and let me know if it looks OK.